pkg/azureutils/azure_disk_utils.go

```diff
@@ -376,8 +382,8 @@ func GetMaxSharesAndMaxMountReplicaCount(parameters map[string]string) (int, int
 		}
 	}
 
-	if maxShares-1 < maxMountReplicaCount {
-		klog.Warningf("maxMountReplicaCount cannot be larger than maxShares - 1: defaulting to maxShares - 1")
+	if maxShares-1 < maxMountReplicaCount || maxMountReplicaCount < 0 {
```

Comment:
If `maxShares` is `0`, then `maxMountReplicaCount` will be set to `-1` again and returned from this function. We should be defensive against this condition and at the very least return `maxShares = 1` and `maxMountReplicaCount = 0`. Since these are user-specified parameters, we should consider returning an error from this function when incorrectly configured.

Reply:
Good point. I will add a check to prevent such scenario.

Reply:
For now, I have added warning statements and went ahead to defaulting maxShares and maxMountReplica to `1` and `maxShares - 1` respectively when incorrectly configured. We can discuss whether we should return error for such or proceed with default values with supplementary documentation for such behavior.